File: C++算法编程课C1/c++练习/OJ练习/3227.cpp
#include<iostream>
#include<iomanip>
#include<algorithm>
#define MAXN 1005 
using namespace std;
struct paidui{
	int value;
	int p;
};
paidui a[MAXN];
bool cmp(const paidui &a, const paidui &b)
{
	if(a.value == b.value)//时间相同，先来的先排序 
		return a.p < b.p;
	else
		return a.value < b.value;//否则按照花费时间从小到大排序 
}
int main()
{
	int n;
	cin >> n;
	for(int i = 1; i <=n; ++i)
	{
		cin >> a[i].value;
		a[i].p = i;
	}
	sort(a+1,a+1+n,cmp);
	double sum=0;//必须用double
	for(int i = 1; i <= n; ++i)
	{
		cout << a[i].p << " ";
		sum += (n-i)*a[i].value; 
		//cout << "sum = " << sum << endl;
	}
	cout << endl;
	cout << fixed << setprecision(2) << sum/n << endl;
	return 0; 
} 

File: C++算法编程课C1/c++练习/OJ练习/3227.py
class Paidui:
    def __init__(self, value, p):
        self.value = value
        self.p = p

n = int(input())
values = list(map(int, input().split()))
a = []
for i in range(1, n+1):
    a.append(Paidui(values[i-1], i))

a.sort(key=lambda x: (x.value, x.p))

sum = 0
for i in range(1, n+1):
    #print(a[i-1].p, end=" ")
    sum += (n-i) * a[i-1].value

#print()
print(f"{sum/n:.2f}")